assets/scripts/routes/common.js

```diff
@@ -14,38 +12,42 @@ export default {
 					sections[i].className += ' focusable';
 				}
 
-				// If there is a '#' in the URL (someone linking directly to a page with an anchor), go directly to that area and focus is
-				// Thanks to WebAIM.org for this idea
-				if ( document.location.hash && document.location.hash !== '#' ) {
-					let anchorUponArrival = document.location.hash;
-					setTimeout( function () {
-						$( anchorUponArrival ).scrollTo( { duration: 1500 } );
-						$( anchorUponArrival ).trigger( 'focus' );
-					}, 100 );
+				// Scroll to anchor if there's a hash in the URL & anchor exists on the page
+				if (document.location.hash && document.location.hash !== '#') {
+					const anchorUponArrival = document.location.hash;
+					setTimeout(() => {
+						const $anchorElement = $(anchorUponArrival);
+						if ($anchorElement.length) {
+							$anchorElement.scrollTo({ duration: 1500 });
+							$anchorElement.trigger('focus');
+						} else {
+							console.warn(`Anchor element "${anchorUponArrival}" not found in the document.`);
```

Comment:
It could fail, silently, sure. I was adding it in case someone wanted to understand why the page didn't jump to the expected anchor point. Previously we were just throwing an error if the anchor wasn't present: https://github.com/pressbooks/pressbooks-aldine/issues/235. Happy to defer to your judgment about what's best

Reply:
got it. I think it would be good to debug, maybe locally or on dev, but I am not sure if we want to print errors in the console in production for debugging purposes. In case we want to debug it in prod, maybe we could think of some strategy, such as sending it somewhere in the backend and putting it in the error logs?

Reply:
I wasn't intended it to be used for debugging, but to inform the browser user. If, for example, as a reader I pasted a link that looks like this: `https://pressbooks.pub/mybook/#fake` but the fake anchor doesn't exist in the page, I might be surprised that the page doesn't load at the expected anchor. Opening the browser console would display a warning message telling me that the anchor `fake` doesn't exist on the page so I could correct or update the URL path.